shanoir-uploader/src/main/java/org/shanoir/uploader/action/RSDocumentListener.java

```diff
@@ -14,33 +14,19 @@ public RSDocumentListener(final MainWindow mainWindow) {
 	}
 
 	public void insertUpdate(DocumentEvent e) {
-		if (mainWindow.noAnonR.isSelected()) {
-			if (mainWindow.lastNameTF.getText().length() != 0 & mainWindow.firstNameTF.getText().length() != 0
-					& mainWindow.birthNameTF.getText().length() != 0 & mainWindow.birthDateTF.getText().length() != 0
-					& (mainWindow.mSexR.isSelected() || mainWindow.fSexR.isSelected())
-					& mainWindow.isDicomObjectSelected == true)
-				mainWindow.downloadOrCopyButton.setEnabled(true);
-		} else {
-			if (mainWindow.newPatientIDTF.getText().length() != 0 & mainWindow.birthDateTF.getText().length() != 0
-					& (mainWindow.mSexR.isSelected() || mainWindow.fSexR.isSelected())
-					& mainWindow.isDicomObjectSelected == true)
-				mainWindow.downloadOrCopyButton.setEnabled(true);
-		}
+		if (mainWindow.lastNameTF.getText().length() != 0 & mainWindow.firstNameTF.getText().length() != 0

```

Comment:
I think this long condition should be stored in a well named variable (e.g. `isFormComplete`)